src/commands/stac-setup/stac.setup.ts

```diff
@@ -0,0 +1,196 @@
+    } else {
+      const metadata: SlugMetadata = {
+        geospatialCategory: args.geospatialCategory,
+        region: args.region,
+        geographicDescription: args.geographicDescription,
+        date: args.addDateInSlug ? formatDate(args.startDate, args.endDate) : '',
+        gsd: args.gsd,
+      };
+      const slug = generateSlug(metadata);
+      const collectionId = ulid.ulid();
+      await writeSetupFiles(startTime, slug, collectionId, args.output);
+    }
+  },
+});
+
+/**
+ * Generates slug based on dataset category.
+ *
+ * @param metadata
+ * @returns slug
+ */
+export function generateSlug(metadata: SlugMetadata): string {
+  const geographicDescription = metadata.geographicDescription || metadata.region;
+  const slug = slugify(metadata.date ? `${geographicDescription}_${metadata.date}` : geographicDescription);
+
+  if (
+    [
+      dataCategories.AERIAL_PHOTOS,
+      dataCategories.RURAL_AERIAL_PHOTOS,
+      dataCategories.SATELLITE_IMAGERY,
+      dataCategories.URBAN_AERIAL_PHOTOS,
+    ].includes(metadata.geospatialCategory)
+  ) {
+    return `${slug}_${metadata.gsd}m`;
+  }
+  if ([dataCategories.DEM, dataCategories.DSM].includes(metadata.geospatialCategory)) {
+    return `${slug}`;
+  }
+  if (metadata.geospatialCategory === dataCategories.SCANNED_AERIAL_PHOTOS) {
+    // nb: Historic Imagery is out of scope as survey number is not yet recorded in collection metadata
+    throw new Error(`Automated slug generation not implemented for historic imagery.`);
+  }
+  throw new Error(`Slug can't be generated from collection as no matching category: ${metadata.geospatialCategory}.`);
+}
+
+/**
+ * Format a STAC collection as a "startYear-endYear" or "startYear" in Pacific/Auckland time
+ *
+ * @param startDate start capture date
+ * @param endDate end capture date
+ * @returns the formatted slug dates
+ */
+export function formatDate(startDate: string, endDate: string): string {
+  const startYear = startDate.slice(0, 4);
+  const endYear = endDate.slice(0, 4);
+
+  if (startYear == null || endYear == null) throw new Error(`Missing datetime field`);
+  if (startYear === endYear) return startYear;
+  return `${startYear}-${endYear}`;
+}
+
+/**
+ * Write the STAC setup values to files for Argo to use
+ *
+ * @param startTime start time for logging information
+ * @param slug the STAC linz:slug value to write
+ * @param collectionId the STAC collection ID value to write
+ * @param output the output path for the setup files
+ */
+export async function writeSetupFiles(
+  startTime: number,
+  slug: string,
+  collectionId: string,
+  output?: URL,
+): Promise<void> {
+  const slugPath = new URL('linz-slug', output);
+  const collectionIdPath = new URL('collection-id', output);
+  await fsa.write(urlToString(slugPath), slug);
+  await fsa.write(urlToString(collectionIdPath), collectionId);
+  logger.info({ duration: performance.now() - startTime, slug, collectionId }, 'StacSetup:Done');
```

Comment:
I think I'd prefer if this was logged in the caller, so that we don't have to pass in the irrelevant `startTime` parameter.